pallets/funding/src/tests/1_application.rs

```diff
@@ -0,0 +1,1321 @@
+			let mut project_metadata = default_project_metadata(inst.get_new_nonce(), ISSUER_1);
+
+			let project_id = inst.create_new_project(project_metadata.clone(), ISSUER_1);
+
+			project_metadata.minimum_price = PriceOf::<TestRuntime>::from_float(1.0);
+			project_metadata.total_allocation_size = 999u128;
+
+			let jwt = get_mock_jwt(ISSUER_1, InvestorType::Institutional, generate_did_from_account(ISSUER_1));
+			inst.execute(|| {
+				assert_noop!(
+					Pallet::<TestRuntime>::edit_project(
+						RuntimeOrigin::signed(ISSUER_1),
+						jwt.clone(),
+						project_id,
+						project_metadata.clone()
+					),
+					Error::<TestRuntime>::FundingTargetTooLow
+				);
+			});
+
+			project_metadata.minimum_price = PriceOf::<TestRuntime>::from_float(0.00000000001);
+			project_metadata.total_allocation_size = 99999999999999u128;
+			inst.execute(|| {
+				assert_noop!(
+					Pallet::<TestRuntime>::edit_project(
+						RuntimeOrigin::signed(ISSUER_1),
+						jwt,
+						project_id,
+						project_metadata
+					),
+					Error::<TestRuntime>::FundingTargetTooLow
+				);
+			});
+		}
+	}
+}
+
+#[cfg(test)]
+mod remove_project_extrinsic {
+	use super::*;
+
+	#[cfg(test)]
+	mod success {
+		use super::*;
+
+		#[test]
+		fn normal_remove() {
+			let mut inst = MockInstantiator::new(Some(RefCell::new(new_test_ext())));
+			let project_metadata = default_project_metadata(inst.get_new_nonce(), ISSUER_1);
+			inst.mint_plmc_to(default_plmc_balances());
+			let jwt = get_mock_jwt(ISSUER_1, InvestorType::Institutional, generate_did_from_account(ISSUER_1));
+			let project_id = inst.create_new_project(project_metadata.clone(), ISSUER_1);
+			assert_ok!(inst.execute(|| crate::Pallet::<TestRuntime>::remove_project(
+				RuntimeOrigin::signed(ISSUER_1),
+				jwt.clone(),
+				project_id
+			)));
+			inst.execute(|| {
+				assert!(ProjectsDetails::<TestRuntime>::get(project_id).is_none());
+				assert!(ProjectsMetadata::<TestRuntime>::get(project_id).is_none());
+				assert!(Buckets::<TestRuntime>::get(project_id).is_none());
+				assert!(DidWithActiveProjects::<TestRuntime>::get(generate_did_from_account(ISSUER_1)).is_none());
+			});
+
+			// removing when no off-chain hash was set works too
+			let mut no_hash_project_metadata = project_metadata.clone();
+			no_hash_project_metadata.offchain_information_hash = None;
+			let project_id = inst.create_new_project(no_hash_project_metadata, ISSUER_1);
+			assert_ok!(inst.execute(|| crate::Pallet::<TestRuntime>::remove_project(
+				RuntimeOrigin::signed(ISSUER_1),
+				jwt.clone(),
+				project_id
+			)));
+			inst.execute(|| {
+				assert!(ProjectsDetails::<TestRuntime>::get(project_id).is_none());
+				assert!(ProjectsMetadata::<TestRuntime>::get(project_id).is_none());
+				assert!(Buckets::<TestRuntime>::get(project_id).is_none());
+				assert!(DidWithActiveProjects::<TestRuntime>::get(generate_did_from_account(ISSUER_1)).is_none());
+			});
```

Comment:
Not sure if this makes sense, as we do not care about an off-chain hash in the removal process, right?